### PR TITLE
Fix LRP Support for Dropout Modules

### PR DESCRIPTION
**Pull request by vivekmig**
This fixes LRP to work appropriately with nn.Dropout modules, since dropout backward hooks add an additional grad_input element of None which is not supported with the current backward_hook_activation.

### CODE REVIEW

**Review by NarineK — APPROVED** [review]
Looks great thank you for the fix. It can be that the behavior of backward hook will change in the future releases of PT. We might want to keep eye on it for future releases. 
https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.register_backward_hook
If can be that `register_full_backward_hook` won't have that issue any more.

**Comment by vivekmig** [comment]
> Looks great, thank you for the fix. It can be that the behavior of backward hook will change in the future releases of PT. We might want to keep eye on it for future releases.
> https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.register_backward_hook
> If can be that `register_full_backward_hook` won't have that issue any more.

Yes, you're right, register_full_backward_hook doesn't have the issue anymore, I tested this initially. I wanted to just switch to register_full_backward_hook and avoid adding this special case, but unfortunately because of the issue identified in #659, I postponed that migration and added this condition for now.
